esx_service/utils/auth_data.py

```diff
@@ -576,7 +638,9 @@ def connect(self):
 
         if not need_create_table:
             if os.path.isfile(self.db_path) and self.need_upgrade_db():
-                self.handle_upgrade_db()
+                error = self.handle_upgrade_db()
+                if error:
```

Comment:
As DB changes happen we will want to be able to handle DB versions with separate "handler" code. Right now its the same code thats only able to work with the newest version of the DB. A better approach will be to make the existing code into an interface and a handler that supports the interface for a specific DB version. And we could do the DB upgrade outside of the service code itself.

As users have already started deploying the volume driver and using it for their workloads its impossible to ask them to delete the DB if an upgrade were to fail. A failure to upgrade shouldn't cause the service to disallow using any of the volumes present. Which would be data loss (no access to data) to the user if it did.

May not be for the change done now but going forward we can't take the approach of deleting the DB, the DB is perfectly fine and only the upgrade is failing.